cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -172,17 +168,10 @@ protected void startUp() throws Exception {
       String metricsConfPath;
       String logbackJarName = null;
 
-      // Always copy (or link if local) user requested resources to local node. This is needed because the
-      // driver is running in the same container as the launcher (i.e. this process).
```

Comment:
It is not true and not needed in distributed mode. The not true part is driver is no longer running in the same container as the launcher (client), and hence copying is not needed.